docs/logging.md

```diff
@@ -0,0 +1,204 @@
+To change this, set the following AutoKitteh config value to the lower-case or
+all-caps name of the level you want. For example:
+
+- CLI: `ak config set logger.level debug`
+
+- Config YAML file:
+
+  ```yaml
+  logger:
+    level: debug
+  ```
+
+- Environment variable: `AK_LOGGER__LEVEL=DEBUG`
+
+## Logger Naming
+
+- Local variables: `l` is preferable when its meaning is clear, otherwise
+  `logger` is acceptable too
+
+  - `z` is discouraged - we use Zap, but we don’t want the code to be tied to
+    that assumption
+  - Other combinations of `l` and `logger` are possible for code readability,
+    but not recommended in general
+
+- Struct fields: `logger` or `Logger`
+
+## Logger Initialization
+
+> [!NOTE]
+> TODO: Refactor `extrazap`
+>
+> TODO: Godoc link below
+
+Pass a logger reference (`*zap.Logger`) as a function parameter, or use
+[extrazap](/TODO) to initialize a logger instance:
+
+```go
+func one(l *zap.Logger, ...) {
+    l.Info("...")
+
+    ctx := extrazap.AttachLoggerToContext(context.Background(), l)
+    two(ctx, ...)
+}
+
+func two(ctx context.Context, ...) {
+    l := extrazap.ExtractLoggerFromContext(ctx)
+    l.Info("...")
+}
+```
+
+You can use `extrazap` even if your function doesn't have a context with an
+embedded logger:
+
+```go
+func three() {
+    l := extrazap.ExtractLoggerFromContext(context.Background())
+    l.Info("...")
+}
+```
+
+In other words, you **should** attach a logger to a context before passing it
+to a function that uses `extrazap`, but it will work even if you don't.
+
+Because of that, there's no need to call [zap.L()](https://pkg.go.dev/go.uber.org/zap#L)
+directly.
+
+Anyway, do not pass both a context and a logger as function parameters, this
+is wasteful.
+
+## Log Messages
+
+1. Should be short
+   - Avoid complex or lengthy sentences
+   - Prefer adding details as fields, not in the message string
+2. Should be informative - if the message is too generic or vague, add vital
+   details to it (in addition to adding them as fields)
+3. Should begin with a capital letter
+   - This is unlike Go error strings, but for the
+     [same reason](https://go.dev/wiki/CodeReviewComments#error-strings)
+4. Shouldn’t end with punctuation
+   - This is like Go error strings, but only to support (1) above
+
+## Adding Log Fields to the Logger
+
+As noted above, we always use loggers by reference (`*zap.Logger`), not by
+value (`zap.Logger`).
+
+Because of that, when you call [Logger.With()](https://pkg.go.dev/go.uber.org/zap#Logger.With)
+in a nested code block, redeclare the logger with the `:=` operator:
+
+```go
+l := l.With(...)
+```
+
+Instead of just reassigning it with the `=` operator:
+
+```go
+l = l.With(...)
+```
+
+Whenever possible, this ensures that the added fields don't remain in the
+logger after leaving exiting the code block.
+
+Example:
+
+```go
+func foo(l *zap.Logger) {
+	l.Info("...", zap.String("aaa", "111"))
```

Comment:
```
l.Info(fmt.Sprintf("...", a), zap.String("a", a))
```

->

```
l.Info("... {a} ...", zap.Any("a", a))
```

?

Reply:
@itayd can you clarify what you mean? There's no `Sprintf` for the message in this code snippet, only Zap fields.

Also, I didn't mention it in this doc, but users may consider Zap's `SugaredLogger` if/when they do want to use string formatting in log messages (which is relevant for previous section (`## Log Messages`).

Reply:
yeah sorry. this was a suggestion i made during sync to allow adding "f-string" capability to logging. you can ignore for now.